capeinfra/objectstorage.py

```diff
@@ -4,29 +4,35 @@
 #       aws, gcp, and azure (we're only implementing aws at this time though)
 
 import pulumi_aws as aws
-from pulumi import Archive, Asset, ComponentResource, ResourceOptions
+from pulumi import Archive, Asset, ResourceOptions
 
+from .pulumi import DescribedComponentResource
 
-class VersionedBucket(ComponentResource):
+
+class VersionedBucket(DescribedComponentResource):
     """An object storage location with versioning turned on."""
 
-    def __init__(self, name, opts=None):
+    def __init__(self, name, **kwargs):
         # This maintains parental relationships within the pulumi stack
         super().__init__(
-            "capeinfra:objectstorage:S3VersionedBucket", name, None, opts
+            "capeinfra:objectstorage:S3VersionedBucket",
+            name,
+            **kwargs,
         )
 
-        self.name = f"{name}-bucket"
+        self.name = f"{name}"
 
         self.bucket = aws.s3.BucketV2(
-            f"{self.name}", opts=ResourceOptions(parent=self)
+            f"{self.name}-s3",
+            opts=ResourceOptions(parent=self),
+            tags={"desc-name": f"{self.desc_name} S3 Bucket"},
```

Comment:
```suggestion
            tags={"desc_name": f"{self.desc_name} S3 Bucket"},
```

This should probably be `_` like the others

Reply:
fixed